src/server/cea_608_encoder/cea_608_utility.py

```diff
@@ -56,19 +56,67 @@ def bytes_to_byte_pairs(byte_list: list) -> list:
     return byte_pairs
 
 
-def create_byte_pairs(caption_string: str) -> list:
+def which_char_set(caption_char: str) -> str:
+    """Finds which character set the letter is in
+
+    :param caption_char:
+    :return: the name of the char set the caption_char is in
+    """
+    for char_set_name,list_of_characters in char_sets.items():
+        if caption_char in list_of_characters:
+            return char_set_name
+
+
+def which_channel(channel_toggle: int,char_set: str) -> str:
+    """Provides the correct first byte to a letter depending on the channel toggle
+
+    :param channel_toggle, char_set:
+    :return: The first byte
+    """
+    if channel_toggle == 0:
+        if char_set == 'basic_na_set':
+            return '-1'
+        elif char_set == 'special_na_set':
+            return '11'
+        elif char_set in ('extended_we_sm_set','extended_we_french_set'):
+            return '12'
+        elif char_set in ('extended_we_port_set','extended_we_gd_set'):
+            return '13'
+        #raise ValueError(f'The character set: {char_set} is not supported')
+    elif channel_toggle == 1:
+        if char_set == 'basic_na_set':
+            return '-1'
+        elif char_set == 'special_na_set':
+            return '19'
+        elif char_set in ('extended_we_sm_set','extended_we_french_set'):
+            return '1a'
+        elif char_set in ('extended_we_port_set','extended_we_gd_set'):
+            return '1b'
+        #raise ValueError(f'The character set: {char_set} is not supported')
+    else:
+        raise ValueError(f'Channel toggle must be 0 or 1!')
+
+
+def create_byte_pair(caption_string: str, channel_toggle: int) -> list:
     """Generates a list of byte pairs given a caption string
 
-    :param caption_string:
+    :param caption_string, channel_toggle:
     :return: list of byte pairs
     """
     byte_list = []
     for letter in caption_string:
-        if letter in basic_north_american_char_set:
-            character_hex_value = basic_north_american_char_set[letter]
-            if check_parity(character_hex_value) == 0:
-                character_hex_value = add_parity_to_byte(character_hex_value)
-            byte_list.append(hex(character_hex_value))
+        set_flag = which_char_set(letter)
```

Comment:
What do you think about changing the naming convention for `set_flag`. In my interpretation a flag is something that is turned on/off. Here it can be of various value that we fetch from the character sets.

Reply:
For sure, I'll change it to "char_set_name"